spyder/plugins/editor/widgets/base.py

```diff
@@ -872,8 +873,7 @@ def insert_completion(self, text, completion_position):
                 if cursor.position() != completion_position:
                     return
             # Add text
-            self.insert_text(text)
-            self.document_did_change()
+            self.sig_insert_completion.emit(text)

```

Comment:
Could you explain how things work now when text is going to be inserted to the editor from the completion widget?

I see we only connect to this signal in the snippets extension, which I don't like too much. I mean, perhaps we could change this to something like:

```python
if self.code_snippets:
    self.sig_insert_completion.emit(text)
else:
    self.insert_text(text)
    self.document_did_change()
```

Reply:
I know the the `code_snippets` attribute is not part of this class but we could introduce it in order to make this logic clearer.